include/triton/Conversion/TritonGPUToLLVM/Utility.h

```diff
@@ -564,20 +551,22 @@ void storeDistributedToShared(triton::gpu::MemDescType dstTy,
 SmallVector<Value>
 lowerLdStShared(Location loc, MLIRContext *ctx, LinearLayout cvt,
                 ArrayRef<Value> valsArray, // Input for store, output for load
-                Type llvmElemTy, Value smemBase, Value affineOffset,
+                Type llvmElemTy, Value smemBase,
+                std::function<Value(Value)> smemAddrAddon, Value affineOffset,
                 uint64_t maskSpanAffineOffset,
                 ConversionPatternRewriter &rewriter,
-                const TargetInfoBase &targetInfo);
+                const TargetInfoBase &targetInfo, Operation *op = nullptr);
```

Comment:
Nit: rather than name it as a very generic `op`, what about using a more self-documenting name reflecting what's expected here?